sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/UnusedPrivateMemberTest.cs

```diff
@@ -29,9 +29,637 @@ public class UnusedPrivateMemberTest
     {
         [TestMethod]
         [TestCategory("Rule")]
-        public void UnusedPrivateMember()
+        public void Field_Accessibility()
         {
-            Verifier.VerifyAnalyzer(@"TestCases\UnusedPrivateMember.cs", new UnusedPrivateMember());
+            Verifier.VerifyCSharpAnalyzer(@"
+public class PrivateMembers
+{
+    private int privateField; // Noncompliant {{Remove the unused private field 'privateField'.}}
+//  ^^^^^^^^^^^^^^^^^^^^^^^^^
+    private static int privateStaticField; // Noncompliant
+
+    private class InnerPrivateClass // Noncompliant
+    {
+        internal int internalField; // Noncompliant
+        protected int protectedField; // Noncompliant
+        protected internal int protectedInternalField; // Noncompliant
+        public int publicField; // Noncompliant
+        internal static int internalStaticField; // Noncompliant
+        protected static int protectedStaticField; // Noncompliant
+        protected internal static int protectedInternalStaticField; // Noncompliant
+        public static int publicStaticField; // Noncompliant
+    }
+}
+
+public class NonPrivateMembers
+{
+    internal int internalField;
+    protected int protectedField;
+    protected internal int protectedInternalField;
+    public int publicField;
+    internal static int internalStaticField;
+    protected static int protectedStaticField;
+    protected internal static int protectedInternalStaticField;
+    public static int publicStaticField;
+
+    public class InnerPublicClass
+    {
```

Comment:
What about private and private static fields on inner classes - are they reported on? Same for the corresponding properties test.
Either way, we should have a test case that covers those cases to documented what the expected behaviour is.

Reply:
We don't report internal members other than types. There is no ticket about that and the functionality was not present before and I decided to not increase the scope of the current work. There are test cases for internal fields, properties and methods, even though there are no comments saying why they are not reported.

Reply:
I generally think that reporting internal members should not be part of this rule (because the rule is about private members and internal is not private). I did not check if the other languages report on members with accessibility that corresponds to internal, though.